server/interview/views.py

```diff
@@ -264,38 +371,67 @@ def put(self, request, interview_id):
                 {"detail": "No active interview found."},
                 status=status.HTTP_404_NOT_FOUND
             )
-class MemberInterviewsView(generics.ListAPIView):
-    serializer_class = InterviewSerializer
-    permission_classes = [IsAuthenticated]
 
-    def get_queryset(self):
-        user = self.request.user
-        return Interview.objects.filter(interviewer=user) | Interview.objects.filter(interviewee=user)
-
-class InterviewerInterviewsView(generics.ListAPIView):
+class InterviewDetailViewAsAdmin(generics.RetrieveAPIView):
+    queryset = Interview.objects.all()
     serializer_class = InterviewSerializer
     permission_classes = [IsAuthenticated]
+    lookup_field = 'interview_id'
 
     def get_queryset(self):
-        return Interview.objects.filter(interviewer=self.request.user)
+        return Interview.objects.filter(interview_id=self.kwargs['interview_id'])
 
-class IntervieweeInterviewsView(generics.ListAPIView):
-    serializer_class = InterviewSerializer
-    permission_classes = [IsAuthenticated]
+class InterviewDetailViewAsUser(APIView):
+    permission_classes = [IsAuthenticated, IsInterviewParticipant] 
 
-    def get_queryset(self):
-        return Interview.objects.filter(interviewee=self.request.user)
+    def get(self, request, *args, **kwargs):
+        user = request.user
 
-class InterviewDetailView(generics.RetrieveAPIView):
-    queryset = Interview.objects.all()
-    serializer_class = InterviewSerializer
-    permission_classes = [IsAuthenticated, IsInterviewParticipant]
-    lookup_field = 'interview_id'
+        interviews_data = []        
+        interviews = self._get_user_interviews(user)
 
-    def get_queryset(self):
-        user = self.request.user
+        for interview in interviews:
+            interview_data = self._format_interview_data(interview, user)
+            interviews_data.append(interview_data)
+        
+        return Response(
+            {"interviews": interviews_data},
+            status=status.HTTP_200_OK
+        )
+
+    def _get_user_interviews(self, user):
+        # Retrieve interviews where the user is either interviewer or interviewee
         return Interview.objects.filter(interviewer=user) | Interview.objects.filter(interviewee=user)
 
+    def _format_interview_data(self, interview, user):
+        # Prepare interview data based on user role and interview status
+        behavioral_questions = interview.behavioral_questions.all() if interview.behavioral_questions.exists() else []
+        interview_data = {
+            "interview_id": interview.interview_id,
+            "interviewer": interview.interviewer.username if interview.interviewer else "N/A",
```

Comment:
I think interviewer and interviewee are currently `id`s as opposed to usernames. That's my bad on the poor naming conventions, and we can fix in two future PRs merged at the same time. For now let's change to id

Reply:
change to use serializer and use id on return